### ISSUE TITLE
git lfs client pre-push operation leaves zombies lying around

### ISSUE THREAD

**Issue by rlaakkol** [github/git-lfs#996]
I have a very large repository where I have used BFG repo cleaner to convert files over the entire history to use git lfs. Otherwise this is working nicely, but when I push this bare repository, it will take a very long time (this is not the problem), as I suppose it does this for all (11k) commits:
```
trace git-lfs: run_command: git rev-list --objects ... --not --remotes=origin
trace git-lfs: run_command: git cat-file --batch-check
trace git-lfs: run_command: git cat-file --batch
```

The problem is, that after each trace print like this, 3 more zombies ([git] <defunct>) appear in my process table. And with 11k commits this means that I will run out of PIDs before the operation can finish. I do not know what the processes actually are, but limiting concurrent transfers to 1 did not change the amount of zombie processes generated.

This can probably be considered to be a bug in the client, but if someone comes up with a workaround, I'd be more than happy to try it out. Is there for example a way to limit the git lfs pre-push operation to a subset of commits that need to be handled by it? There is only a handful of commits that actually introduce changes to files that need to be handled by lfs.

**Comment by rlaakkol** [comment]
Just a thought: The startCommand function that is used by revListShas, catFileBatchCheck and catFileBatch uses exec.Start to run the external git command, but exec.Wait may be never called to clean the resources. Does this sound reasonable?

**Comment by rlaakkol** [comment]
I fiddled around a bit and put some cmd.Cmd.Wait()s in convenient places, and no more zombies!

**Comment by technoweenie** [comment]
I haven't heard of or noticed zombie git procs. That's a great analysis of the issue. Do you have a diff or PR for it?

**Comment by rlaakkol** [comment]
I created a PR!

You should probably notice the zombies when pushing multiple commits for a repo that takes sufficiently long to handle by either monitoring `top` or `ps aux | grep "[d]efunct"`

**Comment by technoweenie** [comment]
Just merged #1012. Thanks so much for reporting this, and looking into a fix!
